lib/cn/cn_tree_internal.h

```diff
@@ -186,12 +186,17 @@ struct cn_tree_node {
 void
 cn_node_stats_get(const struct cn_tree_node *tn, struct cn_node_stats *stats);
 
-/* MTF_MOCK */
-bool
-cn_node_isleaf(const struct cn_tree_node *node);
-
-bool
-cn_node_isroot(const struct cn_tree_node *node);
+static inline bool
+cn_node_isleaf(const struct cn_tree_node *tn)
+{
+    return !tn->tn_isroot;
+}
+
+static inline bool
+cn_node_isroot(const struct cn_tree_node *tn)
+{
+    return tn->tn_isroot;
+}
```

Comment:
At some point we should just get rid of cn_node_isleaf(), and use (nodeid == 0) inside of cn_node_isroot().

Reply:
I'd rather have a static inline than use nodeid directly.

Reply:
I agree with Alex. static inline is less mental overhead for readers of the code.

Reply:
I may have not been clear, I was just saying replace tn_isroot with (tn_nodeid == 0), it can remain a static inline since apparently cn_tree_node is visible (I thought these were real functions and cn_tree_node was not visible to all callers).